pkg/bugreport/ingress.go

```diff
@@ -0,0 +1,138 @@
+
+	ingressResourceName        string = "ingresses"
+	ingressBackendResourceName string = "ingressbackends"
+)
+
+func (c *Config) collectIngressReport() error {
+	namespaceList, err := c.KubeClient.CoreV1().Namespaces().List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	for _, namespace := range namespaceList.Items {
+		ingressList, err := c.KubeClient.NetworkingV1().Ingresses(namespace.Name).List(context.Background(), metav1.ListOptions{})
+		if err != nil {
+			return err
+		}
+		for _, ingress := range ingressList.Items {
+			for _, cmd := range ingressReportCommands(namespace.Name, ingress.Name, ingressResourceName) {
+				filename := path.Join(
+					c.rootNamespaceDirPath(),
+					namespace.Name,
+					ingressResourceName,
+					ingress.Name,
+					commandsDirName,
+					strings.Join(cmd, "_"),
+				)
+				if err := runCmdAndWriteToFile(cmd, filename); err != nil {
+					return err
+				}
+			}
+			c.completionSuccess("Collected report for ingress %s/%s", ingress.Namespace, ingress.Name)
+		}
+
+		ingressBackendList, err := c.PolicyClient.PolicyV1alpha1().IngressBackends(namespace.Name).List(context.Background(), metav1.ListOptions{})
+		if err != nil {
+			return err
+		}
+
+		for _, ingressBackend := range ingressBackendList.Items {
+			for _, cmd := range ingressReportCommands(namespace.Name, ingressBackend.Name, ingressBackendResourceName) {
+				filename := path.Join(
+					c.rootNamespaceDirPath(),
+					namespace.Name,
+					ingressBackendResourceName,
+					ingressBackend.Name,
+					commandsDirName,
+					strings.Join(cmd, "_"),
+				)
+				if err := runCmdAndWriteToFile(cmd, filename); err != nil {
+					return err
+				}
+			}
+			c.completionSuccess("Collected report for ingress backend %s/%s", ingressBackend.Namespace, ingressBackend.Name)
+		}
+	}
+
+	return nil
+}
+
+func (c *Config) collectIngressControllerReport() error {
+	ingressControllerLabelsSelectors := []string{
+		nginxIngressControllerLabelSelector,
+		contourIngressControllerLabelSelector,
+	}
+	for _, labelSelector := range ingressControllerLabelsSelectors {
+		if err := c.collectIngressControllerReportByLabelSelector(labelSelector); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (c *Config) collectIngressControllerReportByLabelSelector(labelSelector string) error {
+	namespaceList, err := c.KubeClient.CoreV1().Namespaces().List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	for _, namespace := range namespaceList.Items {
+		podList, err := c.KubeClient.CoreV1().Pods(namespace.Name).List(
+			context.Background(),
+			metav1.ListOptions{LabelSelector: labelSelector},
+		)
+		if err != nil {
+			return err
+		}
+
+		for _, pod := range podList.Items {
+			cmds := append(
+				ingressControllerReportCommands(pod.Namespace, pod.Name),
+				ingressReportCommands(pod.Namespace, pod.Name, "pods")...,
+			)
+			for _, cmd := range cmds {
+				filename := path.Join(
```

Comment:
It seems that we are constructing 3 different file names with similar params using `path.Join`.  I wonder if it'd increase readability to create 3 different functions: `getIngressFilename`, `getIngressBackendFilename`, `getPodFilename` and use these in lieu of `path.Join` - this would hide the common params `c.rootNamespaceDirPath`, `commonsDirName` and `strings.Join(cmd, "_")`. Then the whole command could fit on one row perhaps.